core/cas-server-core-web-api/src/main/java/org/apereo/cas/web/support/WebUtils.java

```diff
@@ -853,6 +853,16 @@ public static Authentication getInProgressAuthentication() {
         return authentication;
     }
 
+    /**
+     * Gets in progress principal for use by scripted attribute repositories.
+     * This method just allows scripts to use WebUtils instead of the longer class that is more likely to change.
+     *
+     * @return the in progress principal
+     */
+    public static Principal getInProgressPrincipal() {
+        return AuthenticationCredentialsThreadLocalBinder.getInProgressPrincipal();
+    }
+
```

Comment:
I am not sure this belongs here; `AuthenticationCredentialsThreadLocalBinder.getInProgressPrincipal()` should be enough, and wrapping it here is not appropriate because it's not a web/webflow utility.